File: aat-lib/src/main/java/ch/bailu/aat_lib/service/IconMapServiceInterface.java
package ch.bailu.aat_lib.service;

import ch.bailu.aat_lib.gpx.GpxPointNode;
import ch.bailu.aat_lib.gpx.interfaces.GpxPointInterface;
import ch.bailu.aat_lib.service.cache.ObjeImageInterface;

public interface IconMapServiceInterface {
    ObjeImageInterface getIconSVG(GpxPointInterface point, int icon_size);

    String toAssetPath(int key, String value);

    String toAssetPath(GpxPointNode gpxPointNode);
}

File: aat-lib/src/main/java/ch/bailu/aat_lib/service/IconMapServiceInterface.kt
package ch.bailu.aat_lib.service

import ch.bailu.aat_lib.gpx.GpxPointNode
import ch.bailu.aat_lib.gpx.interfaces.GpxPointInterface
import ch.bailu.aat_lib.service.cache.ObjImageInterface

interface IconMapServiceInterface {
    fun getIconSVG(point: GpxPointInterface, icon_size: Int): ObjImageInterface?
    fun toAssetPath(key: Int, value: String): String?
    fun toAssetPath(gpxPointNode: GpxPointNode): String?
}
